nixos/modules/system/boot/timesyncd.nix

```diff
@@ -40,7 +40,7 @@ with lib;
     users.users.systemd-timesync.uid = config.ids.uids.systemd-timesync;
     users.groups.systemd-timesync.gid = config.ids.gids.systemd-timesync;
 
-    system.activationScripts.systemd-timesyncd-migration = mkIf (versionOlder config.system.stateVersion "19.09") ''
+    system.activationScripts.systemd-timesyncd-migration = mkIf ((versionOlder config.system.stateVersion "19.09") || !(config.system ? stateVersion)) ''
```

Comment:
This can't work, `stateVersion` is always set because it has a default, and even if it wouldn't be always set `config.system ? stateVersion` would still always be `true`.